packages/core-data/src/hooks/constants.ts

```diff
@@ -0,0 +1,7 @@
+/* eslint-disable-next-line no-shadow */
```

Comment:
Why do we have to disable this rule, a bug in eslint?

Reply:
I think so, see https://stackoverflow.com/questions/63961803/eslint-says-all-enums-in-typescript-app-are-already-declared-in-the-upper-scope

Perhaps we could disable the no-shadow and enable @typescript-eslint/no-shadow at the global level, but that seems beyond the scope of this PR so I went for eslint-disable for now.

Reply:
> Perhaps we could disable the no-shadow and enable @typescript-eslint/no-shadow at the global level, but that seems beyond the scope of this PR so I went for eslint-disable for now.

It isn't the first time we would do use the rule from TS version of ESLint 👍🏻

Reply:
Here's a PR to adjust the eslint rules: https://github.com/WordPress/gutenberg/pull/38665